Wikipedia/Code/WMFArticleViewController.m

```diff
@@ -641,9 +641,9 @@ - (void)saveWebViewScrollOffset {
 
 #pragma mark - Article Fetching
 
-- (void)fetchArticleIfNeeded {
+- (void)fetchArticleForce:(BOOL)force {
     NSAssert(self.isViewLoaded, @"Should only fetch article when view is loaded so we can update its state.");
-    if (self.article) {
+    if (!force && self.article) {
         return;
     }
     [self showProgressViewAnimated:YES];
```

Comment:
:-1: when refreshing an article that's up to date (should be no change), the view flickers and the progress gets stuck at 80% (i.e. 100% download). IMO, for the case where we already have an article we should wait until the revision check finishes before updating the UI. if the article is out of date, show progress UI as usual, otherwise do nothing (for now). ideally, we'd also show some kind of loading indicator (e.g. network activity indicator) and confirm to the user that their article is up to date if revision is current.

![screen shot 2016-02-10 at 3 00 24 pm](https://cloud.githubusercontent.com/assets/444217/12960022/09b7a8aa-d007-11e5-943a-3cd55005bb92.png)